mainapp/views/views.py

```diff
@@ -243,18 +247,51 @@ def file_serve_proxy(
 
 
 def file_serve(request, id):
-    logger.warning("Serving media files through django is slow")
-
     """ Ensure that the file is not deleted in the database """
-    get_object_or_404(File, id=id)
+    file = get_object_or_404(File, id=id)
 
-    minio_file = minio_client().get_object(minio_file_bucket, str(id))
-    response = HttpResponse(minio_file.read())
+    name, ext = splitext(file.filename)
+    if name.isnumeric() and file.name:
+        filename = f'{file.name}_{name}{ext}'
```

Comment:
is file.name save to use here, both regarding used characters and length?

Reply:
Seems like this is not an easy topic.

There is [RFC5987](https://www.rfc-editor.org/rfc/rfc5987) which defines the encoding of UTF-8 strings in Header fields.

Also it seems like the filename needs to be percentage-encoded.

See also: https://stackoverflow.com/questions/93551/how-to-encode-the-filename-parameter-of-content-disposition-header-in-http

I will push a fix in a minute.